contracts/src/gateway/GatewayManagerFacet.sol

```diff
@@ -171,14 +171,15 @@ contract GatewayManagerFacet is GatewayActorModifiers, ReentrancyGuard {
         supplySource.expect(SupplyKind.ERC20);
 
         // Lock the specified amount into custody.
-        supplySource.lock({value: amount});
+        // Account for tokens with Transfer Fees
+        uint256 transferAmount = supplySource.lock({value: amount});
```

Comment:
According to Zokyo, the `transferAmount` should be clamped by the originally requested amount. So if the transfer happens to lock more tokens than desired, we would still mint and transfer the exact amount of tokens requested by the user.

However, this will lead to locked funds:

* The user deposits 100.
* The inflationary/rewarding token ends up transferring 150 to us.
* We only recognise 100 and mint that amount to the user in the subnet.
* We increment the circulating supply by 100, not 150.
* Those extra 50 will not be accounted for in the protocol and they will end up locked forever in treasury.

I'm checking with Zokyo again, because I don't think their recommendation was valid for our protocol.

Reply:
My current understanding based on the latest communications from Zokyo that we are following their recommended protocol by taking the difference of token balances before and after the transfer. Summarized 

```
So the actual amount transferred into the protocol will be actual = after - before
Which should account for both inflationary and deflationary tokens on transfer
```

Does that sound right?

Reply:
Agreed. Sounds right.